src/ElmSharp/ElmSharp/EvasObject.cs

```diff
@@ -419,6 +442,25 @@ public EvasCanvas EvasCanvas
             }
         }
 
+        /// <summary>
+        /// Sets of gets a value that allow the automation framework to find and interact with this object.
+        /// </summary>
+        /// <since_tizen> preview </since_tizen>
+        public string AutomationId
+        {
+            get
+            {
+                return _automationId;
+            }
+            set
+            {
+                if (_automationId != null)
+                    throw new InvalidOperationException("AutomationId may only be set one time.");
+                _automationId = value;
+                Elementary.SendAutomationIdSet(this);
```

Comment:
If you agree with my suggestion, this line should be moved to nearby `OnRealized()`.